crates/nu-protocol/src/completion.rs

```diff
@@ -0,0 +1,62 @@
+use crate::DeclId;
+use crate::Type;
+use crate::engine::CommandType;
+use serde::{Deserialize, Serialize};
+
+// A simple semantics suggestion just like nu_cli::SemanticSuggestion, but it
+// derives `Serialize` and `Deserialize`, so plugins are allowed to use it
+// to provide dynamic completion items.
+//
+// Why define a new one rather than put `nu_cli::SemanticSuggestion` here?
+//
+// If bringing `nu_cli::SemanticSuggestion` here, it brings reedline::Suggestion too,
+// then it requires this crates depends on `reedline`, this is not good because
+// protocol should not rely on a cli relative interface.
+#[derive(Debug, Default, PartialEq, Serialize, Deserialize, Clone)]
+pub struct DynamicSemanticSuggestion {
+    pub suggestion: DynamicSuggestion,
+    pub kind: Option<SuggestionKind>,
+}
+
+impl From<String> for DynamicSemanticSuggestion {
+    fn from(value: String) -> Self {
+        Self {
+            suggestion: DynamicSuggestion {
+                value,
+                append_whitespace: true,
```

Comment:
I don't think it's necessary to append whitespace. Normal string custom completion suggestions aren't treated like this

Reply:
I'd prefer not appending whitespace, but I don't have strong opinions either way, so it's okay